### ISSUE TITLE
Uninstall

### ISSUE THREAD

**Issue by cocotte42** [Rello/analytics#273]
Hi, first of all I want to apologize, I know I don't give you enough information, but I test it from a production server (BAD !! 😅), and the because it make the memory full I cannot do more tests !

The "bug" is : I install v4.2.1 on Nextcloud v22.2.5.
I just tested to add a new datasource from a local xlsx file. Collabora Online - Built-in CODE Server.
After do it I navigated to the file on the interface to check what my xlsx file contain, and came back to the "analytics" page. Then the left side just display the round to wait.... I wait, wait, then find that apache2 had fully use memory of the server !!!

OK, I restart apache2 just in time before be completly stuck. Because I was thinking, it was the datasource that make the problem, I disabled Analytics, desinstall it, then reinstall it. But, the problem is still there !!!

So there is two issue :
1) It seems that my definition of datasource make a infinited loop of sometinh like that, unfortunately I cannot give you more elements...
2) The most annoying thing, it seems that all datas or not wiped when you unistall it... So, the app is not totally stuck for me.

**Comment by Rello** [comment]
Hi,
if there is an infinite loop in the datasource, this will only kick in once you klick on the report.
if you go to the analytics app like that, it is not opening a report yet.

- when opening analytics, can you please check in the browser console, if there is an error?
- in parallel, please check the nextcloud log. there could be another entry for analytics.

let me know.
I am confident that we will find it.

I also used it for local excel files already. the CODE Server is not involved here, because the file is local - CODE is only for editing the file in your browser

**Comment by cocotte42** [comment]
Many thanks for your quick response.
For the "loop", I finally found some logs in the nextcloud.log. The following lines was repeated about 60000 times.
`{"reqId":"EHzQJINzqFJGDYvyqsMl","level":3,"time":"2022-03-30T20:45:59+02:00","remoteAddr":"87.255.151.54","user":"admin","app":"PHP","method":"GET","url":"/apps/analytics/report","message":"Undefined index: id at /var/www/nextcloud/apps/analytics/lib/Service/ReportService.php#125","userAgent":"Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36","version":"22.2.5.1","exception":{"Exception":"Error","Message":"Undefined index: id at /var/www/nextcloud/apps/analytics/lib/Service/ReportService.php#125","Code":0,"Trace":[{"file":"/var/www/nextcloud/apps/analytics/lib/Service/ReportService.php","line":125,"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::","args":[8,"Undefined index: id","/var/www/nextcloud/apps/analytics/lib/Service/ReportService.php",125,{"ownReports":[{"dataloads":"1","schedules":1,"favorite":0}],"dataloads":[{"dataset":"3","dataloads":"1","schedules":"h"}],"dataload":{"dataset":"3","dataloads":"1","schedules":1},"key":false,"sharedReports":[],"0":"And 4 more entries, set log level to debug to see all entries"}]},{"file":"/var/www/nextcloud/apps/analytics/lib/Controller/ReportController.php","line":45,"function":"index","class":"OCA\\Analytics\\Service\\ReportService","type":"->","args":[]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":217,"function":"index","class":"OCA\\Analytics\\Controller\\ReportController","type":"->","args":[]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":126,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Analytics\\Controller\\ReportController"},"index"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":156,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Analytics\\Controller\\ReportController"},"index"]},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":302,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Analytics\\Controller\\ReportController","index",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"analytics.report.index"}]},{"file":"/var/www/nextcloud/lib/base.php","line":1006,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/apps/analytics/report"]},{"file":"/var/www/nextcloud/index.php","line":36,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/nextcloud/lib/private/Log/ErrorHandler.php","Line":92,"CustomMessage":"--"}}
`

For the second question, how can I totally wipe installation ?

**Comment by Rello** [comment]
Hello,
this is strange. 

second question: when disabling apps, Nextcloud does not remove the tables by design. this can only be done with database access.

but: please give me some more questions. I would like to find out what happened here.
You created some special combination which I want to find out.

so

1. I ***ed your domain which you posted above. you might not want to make this public
2. It looks like there is no report existing but datasets. I created this combination, but it works for me
3. can you please go to `https://***/apps/analytics/a/` this will bring you to the dataset maintenance
4. Are you seeing some there? please delete them
5. then when you go back to the initial screen of analytics, is it working again?

do you have a second user to test? I expect that the app will start as normal there.

**Comment by Rello** [comment]
I think I managed to reproduce this partly - but it does not brake on my side.
![Bildschirmfoto 2022-03-30 um 22 35 58](https://user-images.githubusercontent.com/13385119/160926314-d81f26d7-ac53-49c0-9b5c-967bbe28decd.png)

waiting for your hint...

**Comment by cocotte42** [comment]
Right, I was able to delete the dataset with the direct url, then back to inital screen : working fine now.
I tested before with another user, the problem was not there.

For your information, I looked more closely, the memory was not really full because of "Analytics". If fact, it "just" take 100% cpu and generate 60000 lines in the log. The full memory was a side effect on another service that analyse logs and doesn't really like so many lines so quickly 😅 (thanks to help me to find and correct this simple DoS issue !!)

I will try to play with your cool apps, if I reproduce this issue, I will try to remumber more precisely what I have done.
Thanks again !

**Comment by Rello** [comment]
ok. I will put in an additional check for this condition. thank you for finding this.

but the 60k lines. I wonder where they came from. because this part is only executed when starting the app

**Comment by Rello** [comment]
4.3.0 released
